backend/src/main/java/hu/flowacademy/meteo/service/FileUploadService.java

```diff
@@ -0,0 +1,138 @@
+    private static final String DATA_TYPE_HOURLY = "orai";
+
+    private final StationRepository stationRepository;
+    private final MeasurementRepository measurementRepository;
+
+    @Autowired
+    public FileUploadService(StationRepository stationRepository, MeasurementRepository measurementRepository) {
+        this.stationRepository = stationRepository;
+        this.measurementRepository = measurementRepository;
+    }
+
+    public void fileUpload(MultipartFile file, String dataType, String stationName) throws IOException {
+        try (InputStreamReader fileStreamReader = new InputStreamReader(file.getInputStream())) {
+            Optional<Station> station = stationRepository.findFirstByName(stationName);
+            if (station.isPresent()) {
+                log.debug("Repsository size before upload: {}", measurementRepository.count());
+                executeMeasurementSave(station.get(), fileStreamReader, getType(dataType));
+                log.debug("Repsository size after upload: {}", measurementRepository.count());
+            }
+        } catch (IOException e) {
+            log.error("Error while reading file: {}", file.getOriginalFilename(), e);
+            throw e;
+        }
+    }
+
+    public Type getType(String dataType) {
+        return (DATA_TYPE_DAILY.equals(dataType) ? Type.DAILY : DATA_TYPE_HOURLY.equals(dataType) ? Type.HOURLY : Type.TEN_MIN);
+    }
+
+    public Object formatDoubleData(String str) {
+        if (str.equals("") || str.equals("-9999,00000")) {
+            return null;
+        }
+        return Double.parseDouble(str.replace(",", "."));
+    }
+
+    public String dataGetter(String[] data, Map<String, Integer> headerKeys, String header) {
+        return headerKeys.containsKey(header) ? data[headerKeys.get(header)] : "";
+    }
+
+    public void executeMeasurementSave(Station station, Reader reader, Type type) {
+        List<Measurement> measurements = measurementRepository.saveAll(populateDataBase(reader, station, type));
+        log.info("saved {} {} measurments at station: {} with id: {}", measurements.size(), type, station.getName(), station.getId());
+        if (measurements.size() == 0) {
+            throw new ValidationException("File upload failed!");
+        }
+    }
+
+    public Map<String, Integer> getKeys(String firstLine) {
+        String[] keys = firstLine.split(";\\s*", -1);
+        Map<String, Integer> dataMap = new HashMap<>();
+        for (int i = 0; i < keys.length; i++) {
+            dataMap.put(keys[i], i);
+        }
+        return dataMap;
+    }
+
+    public List<Measurement> populateDataBase(Reader reader, Station station, Type type) {
+        String line;
+        List<Measurement> list = new ArrayList<>();
+        try (BufferedReader br = new BufferedReader(reader)) {
+            Map<String, Integer> dataMap = getKeys(br.readLine());
+            log.debug("Found {} keys!!!", dataMap.size());
```

Comment:
Ha talált kulcsok száma 2-nél kevesebb, akkor adat hiba van (biztosan lehet bátrabb állításokat is megfogalmazni), és lehet olyan exception-t dobni, amiből 400-as errort csinál a controller.